### DISCUSSION TITLE
upgrading cpp/c mixed project from 3.2 to 3.4 fail to build.

### DISCUSSION THREAD

**Discussion by eyalsol** [zephyrproject-rtos/zephyr#66527] [Q&A] [answered]
we are trying to upgrade the project, we work on, from ZephyrOS 3.2 to 3.4

the prj.conf in version 3.2:

 ```
 #enable c++ support
CONFIG_CPLUSPLUS=y
CONFIG_LIB_CPLUSPLUS=y
CONFIG_STD_CPP20=y
CONFIG_NEWLIB_LIBC=y
```

the prj.conf in version 3.4:

```
CONFIG_CPP=y
CONFIG_REQUIRES_FULL_LIBCPP=y
CONFIG_REQUIRES_FULL_LIBC=y
CONFIG_STD_CPP20=y

```

The Error we get:
```
in file included from ~/zephyr_workspace/zephyr/include/zephyr/types.h:11,
                 from ~/zephyr_workspace/zephyr/include/zephyr/kernel_includes.h:17,
                 from ~/zephyr_workspace/zephyr/include/zephyr/kernel.h:17,
                 from ~/zephyr_workspace/zephyr/arch/riscv/core/offsets/offsets.c:16:
~/tools/syntacore/sc-dt-2023.11/riscv-gcc/lib/gcc/riscv64-unknown-elf/12.2.1/include/stdint.h:9:16: fatal error: stdint.h: No such file or directory
    9 | # include_next <stdint.h>
      |                ^~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```

we have tried other config parameters but it was still failing with other errors.
Any assistance will be appreciated

Thanks

**Comment by eyalsol** [comment, marked as answer]
Issue resolved.

the toolchain in use is syntacore: sc-dt-2023.11, riscv-gcc, riscv64-unknown-elf.

I had to explicitly set this in CMakeLists.txt:

set(TOOLCHAIN_HAS_NEWLIB ON CACHE BOOL "True if toolchain supports newlib") 
the prj.conf is set to:

CONFIG_CPP=y
CONFIG_REQUIRES_FULL_LIBCPP=y
CONFIG_NEWLIB_LIBC=y
CONFIG_STD_CPP20=y

closing discussion as resolved

**Reply by TorstenRobitzki** [reply]
Thanks, that saved my day :-) Same applies, if you want to use an arm-none-eabi-gcc.

**Reply by Git-qisen** [reply]
Hello, I encountered the same problem when building cpp/c mixed project in Zephyr version 3.7.0. 

I have modified the CMakeList.txt and prj.conf according to the method you provided, but the error of <stdint.h> is still not found. By the way, the compiler I used was riscv64-unknowed-elf. 

I was wondering if there were any other changes involved besides the above, or how to put the external library containing <stdint.h> into the zephyr project

**Reply by TorstenRobitzki** [reply]
Not that I can remember, sorry.
